modules/nosql/manager_redis.go

```diff
@@ -59,8 +60,84 @@ func (m *Manager) GetRedisClient(connection string) redis.UniversalClient {
 		name: []string{connection, uri.String()},
 	}
 
+	opts := getRedisOptions(uri)
+	tlsConfig := getRedisTLSOptions(uri)
+
+	clientName := uri.Query()["clientname"]
+
+	if clientName != nil && len(clientName) > 0 {
+		client.name = append(client.name, clientName[0])
+	}
+
+	switch uri.Scheme {
+	case "redis+sentinels":
+		fallthrough
+	case "rediss+sentinel":
+		opts.TLSConfig = tlsConfig
+		fallthrough
+	case "redis+sentinel":
+		if uri.Host != "" {
+			opts.Addrs = append(opts.Addrs, strings.Split(uri.Host, ",")...)
+		}
+		if uri.Path != "" {
+			if db, err := strconv.Atoi(uri.Path[1:]); err == nil {
+				opts.DB = db
+			}
```

Comment:
could add a `else { print error }` in case users make mistakes with the path.

Reply:
I'm happy to do so if you'd like -- this was all original code just moved out to another method.

Reply:
yes if you can improve old "original" code we should do so